cmake/toolchain/xcc/generic.cmake

```diff
@@ -23,7 +23,7 @@ set(SYSROOT_DIR    ${TOOLCHAIN_HOME}/${SYSROOT_TARGET})
 # xt-xcc does not support -Og, so make it -O0
 set(OPTIMIZE_FOR_DEBUG_FLAG "-O0")
 
-if(EXISTS ${CROSS_COMPILE}clang)
```

Comment:
This (kind of) auto detects if clang is available, and will use clang is it exists. Maybe what you want here is to override the auto detection?

Reply:
right, the question is if CLANG is desired, what if you have clang installed by you do not want to use it and want gcc instead?